9.0 web widget datepicker fix

StephanRozendaal opened the pull request:
Fix #684

StephanRozendaal committed:
[FIX] widget does not set options on the widget

The values of the date widget of the field are not being set.
This is because the start function does not update the options,
instead they are set in the init function. inherit the init
function of the widget, and add the options here.

StephanRozendaal committed:
[REM] Remove inheritance on DateTimeWidget

The inheritance of the start function is not needed, because
the function is already inherited from DateWidget.

yajo commented on web_widget_datepicker_options/static/src/js/datepicker.js:
```diff
@@ -24,34 +24,16 @@ odoo.define('web_widget_datepicker_options', function (require) {
-
     DateWidget.include({
-        start: function(parent, options) {
+	init: function(parent, options) {
```
Use 4 spaces indentation please.

StephanRozendaal replied:
Looks like I used a tab, it's fixed now.

yajo commented:
Could you explain how to test the thing that this is fixing?

StephanRozendaal commented:
@Yajo This module adds two demonstration fields on a res.users form. both fields have " options="{'datepicker':{'calendarWeeks': false}}" " parameters, in the current 9.0 version setting calendarWeeks to true does nothing to the calendar display of the field when you try to pick a date. With my change, when setting calendarWeeks to true, when picking a date it will display extra weeknumbers.

Which leads to the conclusion that options on the field were not being transfered to the JQuery datepicker widget, and with this change they are.

StephanRozendaal committed:
[FIX] Check if date widget properties exist

the options property of the dateWidget does not exist when the
field is rendered in for example a list view.

Check if the options and datepicker properties exist first.

StephanRozendaal commented:
I've fixed that date fields, when viewed in a list view caused a javascript error.
